### DISCUSSION TITLE
Artisan::queue does not support the functionality of queued jobs

### DISCUSSION THREAD

**Discussion by gms8994** [laravel/framework#52753] [Ideas] [answered]
### Laravel Version

10.34.2

### PHP Version

8.2.22

### Description

If I want to `Artisan::queue(Command::class)`, I cannot set a timeout on that job, which means it could fail based on the timeout of the queue runner. Ideally, either the ability to set the timeout inside the command (via `$timeout` or `public function timeout`) or the ability to control it when calling `::queue` (like `->delay()` or `->onQueue()`).

### Steps To Reproduce

Call `Artisan::queue` on a console command and note that the timeout is not set. I cannot currently do `laravel new bug-report` as suggested due to some other issue.

**Comment by hackeresq** [comment]
What version are you on? 

Looks like Artisan::queue() returns a `PendingDispatch` object, which supports those chained objects: 

<img width="970" alt="image" src="https://github.com/user-attachments/assets/2ed72040-d89a-4a1a-a52a-19462beafe77">

This is working for me on Laravel 11.9.

**Reply by gms8994** [reply]
I'm on 10; I don't want to `delay`, I want to control the timeout of the job, which PendingDispatch doesn't seem to support.

**Reply by henzeb** [reply, marked as answer]
you can create your own method to set the timeout property.

**Reply by gms8994** [reply]
I swear I tried this and it didn't work. Thanks!

**Reply by NiroDeveloper** [reply]
How do you set the timeout property? PendingDispatch::$job is a protected variable.
